### PR TITLE
feat: localiser backbeam approaches

### PR DESCRIPTION
**Pull request by tracernz**
Fixes #[issue_no]

## Summary of Changes
- fix: remove erroneous MathUtils.round
  Number.prototype.toFixed` already does the rounding itself, and I'm about to change MathUtils.round...
- fix: make fbw MathUtils.round quantise like msfssdk
  This is much more powerful and paves the way to avoiding importing both.
- feat: add angle normalise to MathUtils
  Add functions to normalise angles to the common +/- 180, 0-360, and radian equivalents.
- fix(fms): back course approach names
  Fixes the naming of LOC back-course approaches in the FMS (BACnn)
- fix(fms): back course related bugs, and add output vars
  Fixed some issues relating to tuning of back courses, and add the output data for the FG to consume (and other systems directly in the meanwhile until the FG knows how).
- fix(isis): loc deviation on back beam
  We never showed the correct deviations when on the backbeam, as the SimVar we use gives deviations close to -180 or 180 depending which side you're on and that is always off-scale. Other places are fixed in a different commit with new features as well.
- feat: port angle function to C++
  Port the angle related functions to C++ utils in preparation for AP..
- feat(nd): backbeam loc
  Implement backbeam localiser support on the ND ROSE LS page. When a backbeam is tuned the glideslope scale is hidden, a B/C reminder is shown on the left of the localiser scale, and the localiser deviations are inverted. The B/C reminder is replaced by a LOC reminder for LOC/F-GS approaches so prepare for that too when I complete the MMR. Note that the B/C reminder positioning may be a little off as I did not yet have a photograph of a real ND to check. I will fix this in future as needed.
- feat(pfd): loc backbeam, refactor to modern framework
  Implement backbeam localiser support on the PFD. When a backbeam is tuned the glideslope scale is hidden, a B/C reminder is shown on the left of the localiser scale, and the localiser deviations are inverted. The B/C reminder is replaced by a LOC reminder for LOC/F-GS approaches so prepare for that too when I complete the MMR. I refactored some of the parts I touched to modern framework style, fixing a couple of bugs with missing subs in the process and turning the mapping functions into pure functions.
- fix(pfd): heading tape ls indications
  Fixed a bug where the heading tape was listening for the LS button H event directly, rather than using the bus data. This caused it to become out of sync with the actual state in some cases.
- feat(ap): loc backbeam
  Add some workarounds in the FBW interface to allow the AP to fly localiser back courses.

## Screenshots (if necessary)

image
Note ISIS LOC deviation is uncorrected:
image

## Additional context

Discord username (if different from GitHub):

## Testing instructions

- Fly a couple of back-course approaches and ensure the AP can capture and fly the LOC in LOC B/C mode. There is a list below, but they are not filtered by runway length aircraft category so some may be no good for the A320. Check that the deviations on the PFD LS scale and ND ROSE LS page show a B/C reminder flag and invert the deviation so the indicator shows the correct direction to capture the localiser. The ISIS should not have the corrected deviation, and instead be inverted, as it does not have an FMGC input bus to know when a backbeam is tuned. Note: the AP will not be able to track the LOC when you get too close to the runway/localiser, as the localiser transmitter is not at the far end of the runway like a normal LOC approach; it should remain stable until the published minima.
- Fly a regular front-course ILS approach with AP and ensure it works fine and the LOC deviations are on the correct side as normal.

Airport | Approach
-- | --
CYOD | BAC13L
CYYR | BAC26-Y
CYYR | BAC26-Z
KALO | BAC30
KBMI | BAC11
KBRO | BAC31
KCGI | BAC28
KCLL | BAC17
KCMX | BAC14
KDEC | BAC24
KEAU | BAC04
KESC | BAC28
KEWB | BAC23
KFAY | BAC22
KGFK | BAC17R
KGRB | BAC24
KGRI | BAC17
KHON | BAC30
KHRL | BAC35L
KHUT | BAC31
KIDA | BAC03
KIMT | BAC19
KJMS | BAC13
KLBB | BAC35L
KLRD | BAC36L
KMCE | BAC12
KMLB | BAC27L
KMOT | BAC13
KOSH | BAC18
KOTM | BAC13
KPTK | BAC27L
KRDD | BAC16
KRFD | BAC19
KROW | BAC03
KSLE | BAC13
KSNA | BAC02L
KSQI | BAC07
KSTJ | BAC17
KTOP | BAC31
KUIN | BAC22
PABR | BAC26
PACD | BAC33
PAHO | BAC22
PAKN | BAC30
PAOM | BAC10
PASC | BAC24
PASN | BAC18
PAYA | BAC29
PHKO | BAC35

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

File: fbw-a32nx/src/systems/instruments/src/PFD/FMA.tsx
Comment (Saschl) [inline fbw-a32nx/src/systems/instruments/src/PFD/FMA.tsx:1]: FMA looks so much better now :D

**Review by Saschl — APPROVED** [review]
LGTM

**Comment by tracernz** [comment]
Rebased, and fixed a couple of fms-v2 bugs while at it in LandingSystemSelectionManager, and FlightPlan.

**Review by Benjozork — APPROVED** [review]

**Comment by tracernz** [comment]
Rescued a commit from the reflog that I somehow dropping in an interactive rebase 😬 .

**Comment by Saschl** [comment]
QA Report

Tester: saschl

Testing Process:

Tested several programmed Backcourse approaches, ND and PFD show proper indications and AP follows the B/C LOC properly.
Tested manually entered LOC Backcourse (LOWI OEJ), Flags were shown properly and AP was tracking as expected.

Media:

Screenshot 2024-06-25 202022

Screenshot 2024-06-25 202025

Testing Results:
Passed

**Pull request merged**